internal/transport/transport_test.go

```diff
@@ -2670,7 +2745,7 @@ func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
 	greetDone := make(chan struct{})
 	// errorCh verifies that desired GOAWAY not received by server
 	errorCh := make(chan error)
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	ctx, cancel := context.WithTimeout(context.Background(), 2*defaultTestTimeout)
```

Comment:
let's avoid increasing timeouts and try to keep the default timeout. One quick approach we can do here is overriding the timeout interval to 0 for this test and hang the net conn write forever. Then we will always timeout on write because of timer and test will finish quickly closing the client without sending goaway frame